src/log/builder.cr

```diff
@@ -61,11 +61,12 @@ class Log::Builder
   end
 
   # :nodoc:
-  def each_log
+  private def each_log
+    @logs.reject! { |_, log_ref| log_ref.value.nil? }
```

Comment:
Can't we do this deletion while iterating? I'd much rather do it in one loop if possible.

Reply:
If it would be an Array I would say that iterating backwards should work (or playing with indices). I would like mutating iterators :-P

Since it is a Hash, iterating it and mutating it is not a nice option. So either as is, or have a copy of all the keys to iterate.

I ended up with the simpler code.

Reply:
I think this is fine. If it shows to be a performance issue, we can improve it later.